app/models/name/change.rb

```diff
@@ -19,9 +19,9 @@ def change_text_name(in_text_name, in_author, in_rank, save_parents = false)
       parents = Name.find_or_create_name_and_parents(parse.parent_name)
       if parents.last.nil?
         raise(:runtime_unable_to_create_name.t(name: parse.parent_name))
-      elsif save_parents
-        parents.each { |n| n.save if n&.new_record? }
       end
+
+      parents.each { |n| n.save if n&.new_record? } if save_parents
```

Comment:
Does this work? (Same issue with `raise`)

Reply:
I suspect so.  Name should be well-tested, so I would trust this if it passes test.

Reply:
Thanks again @pellaea. The code will work. (I was unsure about this because I didn't understand `raise`, which you have now explained.)